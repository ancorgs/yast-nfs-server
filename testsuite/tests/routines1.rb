# encoding: utf-8

# Module:
#   NFS server configuration
#
# Summary:
#   Routines testsuite
#
# Authors:
#   Martin Vidner <mvidner@suse.cz>
#
# $Id$
module Yast
  class Routines1Client < Client
    def main
      # testedfiles: routines.ycp Testsuite.ycp

      Yast.include self, "testsuite.rb"
      Yast.include self, "nfs_server/routines.rb"

      @exports = [
        { "allowed" => ["proj*.local.domain(rw)"], "mountpoint" => "/projects" },
        {
          "allowed"    => ["*.local.domain(ro)", "@trusted(rw)"],
          "mountpoint" => "/usr"
        },
        { "allowed" => ["(ro,insecure,all_squash)"], "mountpoint" => "/pub" }
      ]

      TEST(lambda { AllowedToHostsOpts("host.com(options)") }, [], nil)
      TEST(lambda { AllowedToHostsOpts("noopts.host.com") }, [], nil)
      TEST(lambda { AllowedToHostsOpts("host.com(noclose") }, [], nil)
      TEST(lambda { AllowedToHostsOpts("host.com((double))") }, [], nil)
      TEST(lambda { AllowedToHostsOpts("") }, [], nil)

      TEST(lambda { AllowedTableItems(["*.local.domain(ro)", "@trusted(rw)"]) }, [], nil)
      TEST(lambda { AllowedTableItems([]) }, [], nil)

      TEST(lambda { FindAllowed(@exports, "/pub") }, [], nil)
      TEST(lambda { FindAllowed(@exports, "/nosuchpath") }, [], nil)
      TEST(lambda { FindAllowed([], "/pub") }, [], nil)

      TEST(lambda { ExportsItems(@exports) }, [], nil)
      TEST(lambda { ExportsItems([]) }, [], nil)

      TEST(lambda { ExportsSelBox(@exports) }, [], nil)
      TEST(lambda { ExportsSelBox([]) }, [], nil)

      TEST(lambda { ReplaceInExports(@exports, "/usr", ["*.localdomain(ro)"]) }, [], nil)
      TEST(lambda do
        ReplaceInExports(@exports, "/nosuchpath", ["*.localdomain(ro)"])
      end, [], nil)
      TEST(lambda { ReplaceInExports([], "/whatever", ["*.localdomain(ro)"]) }, [], nil)

      nil
    end
  end
end

Yast::Routines1Client.new.main
