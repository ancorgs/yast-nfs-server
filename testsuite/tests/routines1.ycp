/**
 * Module:
 *   NFS server configuration
 *
 * Summary:
 *   Routines testuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    include "testsuite.ycp";
    include "nfs_server/routines.ycp";

    list exports = [$["allowed":["proj*.local.domain(rw)"], 
		      "mountpoint":"/projects"], 
		    $["allowed":["*.local.domain(ro)", "@trusted(rw)"], 
		      "mountpoint":"/usr"], 
		    $["allowed":["(ro,insecure,all_squash)"], 
		      "mountpoint":"/pub"]];

    TEST (``(AllowedToHostsOpts ("host.com(options)")), [], nil);
    TEST (``(AllowedToHostsOpts ("noopts.host.com")), [], nil);
    TEST (``(AllowedToHostsOpts ("host.com(noclose")), [], nil);
    TEST (``(AllowedToHostsOpts ("host.com((double))")), [], nil);
    TEST (``(AllowedToHostsOpts ("")), [], nil);

    TEST (``(AllowedTableItems (["*.local.domain(ro)", "@trusted(rw)"])), [], nil);
    TEST (``(AllowedTableItems ([])), [], nil);

    TEST (``(FindAllowed (exports, "/pub")), [], nil);
    TEST (``(FindAllowed (exports, "/nosuchpath")), [], nil);
    TEST (``(FindAllowed ([], "/pub")), [], nil);

    TEST (``(ExportsItems (exports)), [], nil);
    TEST (``(ExportsItems ([])), [], nil);

    TEST (``(ExportsSelBox (exports)), [], nil);
    TEST (``(ExportsSelBox ([])), [], nil);

    TEST (``(ReplaceInExports (exports, "/usr", ["*.localdomain(ro)"])), [], nil);
    TEST (``(ReplaceInExports (exports, "/nosuchpath", ["*.localdomain(ro)"])), [], nil);
    TEST (``(ReplaceInExports ([], "/whatever", ["*.localdomain(ro)"])), [], nil);
}
