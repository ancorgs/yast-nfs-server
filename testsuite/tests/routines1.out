Return	["host.com", "options"]
Return	["noopts.host.com", ""]
Return	["host.com", "noclose"]
Return	["host.com", "(double"]
Return	["", ""]
Return	[`item (`id (0), "*.local.domain ", "ro "), `item (`id (1), "@trusted ", "rw ")]
Return	[]
Return	["(ro,insecure,all_squash)"]
Return	nil
Return	nil
Return	[`item (`id ("/projects"), "/projects "), `item (`id ("/usr"), "/usr "), `item (`id ("/pub"), "/pub ")]
Return	[]
Return	`SelectionBox (`id (`exportsbox), `opt (`notify), "Dire&ctories", [`item (`id ("/projects"), "/projects "), `item (`id ("/usr"), "/usr "), `item (`id ("/pub"), "/pub ")])
Return	`SelectionBox (`id (`exportsbox), `opt (`notify), "Dire&ctories", [])
Return	[$["allowed":["proj*.local.domain(rw)"], "mountpoint":"/projects"], $["allowed":["*.localdomain(ro)"], "mountpoint":"/usr"], $["allowed":["(ro,insecure,all_squash)"], "mountpoint":"/pub"]]
Return	[$["allowed":["proj*.local.domain(rw)"], "mountpoint":"/projects"], $["allowed":["*.local.domain(ro)", "@trusted(rw)"], "mountpoint":"/usr"], $["allowed":["(ro,insecure,all_squash)"], "mountpoint":"/pub"]]
Return	[]
